### PR TITLE
Create Binary_Insertion_Sort.cs

### LINKED ISSUE #2125: Binary Insertion Sort
- [x] C
- [x] C++
- [x] C# - @Yashank18
- [ ] CoffeeScript
- [ ] Dart - @raksha009
- [ ] Go - @KamalDGRT
- [ ] Java - @raktim00
- [ ] JS - @sheetalssr
- [ ] Kotlin
- [ ] Php - @goyalmuskan
- [x] Python
- [ ] Readme - @goyalmuskan
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Yashank18**
Fixes #2125 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implemented Binary Insertion Sort

#### Languages Used:
- C#

#### Files Added:
- Binary_Insertion_Sort.cs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Yashank18** [comment]
@singh-shreya6 @jainaman224

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+using System;
```
Comment (singh-shreya6) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: Please add problem desc at the start of file

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        for (int i = 1; i < array.Length; i++) 
+        { 
+            int x = array[i]; 
+  
+            // Find location to insert using 
+            // binary search 
+            int j = Math.Abs(Array.BinarySearch( array, 0, i, x) + 1); 
+  
+            // Shifting array to one location right 
+            System.Array.Copy(array, j, array, j+1, i-j);
```
Comment (singh-shreya6) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: add space round operators

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        } 
+    }
+    public static void Main() 
+    { 
+        int size;
+        size= Convert.ToInt32(Console.ReadLine()); 
+
+        int[] array = new int[size];
+
+        for(int i=0; i<size; i++)
```
Comment (singh-shreya6) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: give space after for keyword and around operators

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	    {
+	        array[i] = Convert.ToInt32(Console.ReadLine());
+	    }
+
+        sort(array); 
+  
+        for(int i = 0; i < array.Length; i++)
```
Comment (singh-shreya6) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: space after for

**Comment by Yashank18** [comment]
@singh-shreya6 Updated

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/*We can use binary search to reduce the number of comparisons in normal insertion sort. Binary Insertion Sort uses binary search to find the proper location to insert the selected item at each iteration.*/
```
Comment (singh-shreya6) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: Please split the comment to span multiple lines as it is going out of screen currently. Try not to exceed 80 characters per line.

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    public static void Main() 
+    { 
+        int size;
+        size= Convert.ToInt32(Console.ReadLine()); 
+
+        int[] array = new int[size];
+
+        for (int i = 0; i < size; i++)
+	    {
```
Comment (singh-shreya6) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: format these brackets correctly.

**Comment by Yashank18** [comment]
@singh-shreya6 Done

**Comment by Yashank18** [comment]
@singh-shreya6  can you please add Label to it

**Pull request merged**